instruments/abstract_instruments/comm/usb_communicator.py

```diff
@@ -77,25 +111,47 @@ def close(self):
         """
         Shutdown and close the USB connection
         """
-        try:
-            self._conn.shutdown()
-        finally:
-            self._conn.close()
+        self._dev.reset()
+        usb.util.dispose_resources(self._dev)
 
-    def read_raw(self, size=-1):
-        raise NotImplementedError
+    def read_raw(self, size=1000):
+        """Read raw string back from device and return.
 
-    def read(self, size=-1, encoding="utf-8"):
-        raise NotImplementedError
+        String returned is most likely shorter than the size requested. Will
+        terminate by itself.
+        Read size of -1 will be transformed into 1000.
 
-    def write_raw(self, msg):
+        :param size: Size to read in bytes
+        :type size: int
         """
-        Write bytes to the raw usb connection object.
+        if size == -1:
+            size = 1000
```

Comment:
This is really ugly... but makes it compatible with other communicators and the default chosen here.

Reply:
Maybe `wMaxPacketSize` ?

Reply:
Great idea. I don't have the picomotor controller around at the moment, but read the `ENDPOINT_IN` of another, random USB device laying around and can query it for `wMaxPacketSize`. Instead of `size=1000`, I'm using the package size now.
Glad I wrote the tests before...